src/main/java/net/wimpi/modbus/net/SerialConnection.java

```diff
@@ -232,20 +231,19 @@ protected void setConnectionParameters() throws Exception {
             m_Parameters.setStopbits(oldStopbits);
             m_Parameters.setParity(oldParity);
             final String errMsg = "Unsupported parameter";
-            logger.debug("{} failed to set up one of [baudRate, dataBits, stopBits, parity]: {}", errMsg,
-                    e.getMessage());
+            logger.error("{}, failed to set up one of [baudRate, dataBits, stopBits, parity]", errMsg, e);
 
-            throw new Exception(errMsg);
+            throw new Exception(errMsg + ": " + e.getMessage());
```

Comment:
This part of the actual logged error message will be omitted, even though that provides valuable extra information: `, failed to set up one of [baudRate, dataBits, stopBits, parity]`. I'd make that part of the `errMsg` variable itself and thus simplify logger statement to just `logger.error(errMsg, e);`.

Reply:
Good idea, done.

Reply:
OK